ui/src/app/utils/extractPowerType.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Get the formatted power type from a power type description.
+ * @param description - A power type description.
+ * @param powerType - A power type.
+ * @return The formatted power type or the original power type key.
+ */
+export const extractPowerType = (
```

Comment:
I've left the original in the shared workspace for now so it can be used by legacy.